lara-typescript/src/page-item-authoring/common/components/interactive-authoring-preview.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import { useRef, useState } from "react";
+
+import { InteractiveIframe } from "./interactive-iframe";
+
+export interface IPreviewInteractive {
+  url: string;
+  aspect_ratio_method: string;
+  authored_state: string | object;
+};
+
+interface Props {
+  interactive: IPreviewInteractive;
+}
+
+export const InteractiveAuthoringPreview: React.FC<Props> = ({interactive}) => {
+  const iframe = useRef<HTMLIFrameElement|null>(null);
+  const [authoredState, setAuthoredState] = useState<object|null>(
+    typeof interactive.authored_state === "string"
+      ? JSON.parse(interactive.authored_state || "{}")
+      : interactive.authored_state
+  );
+
+  const handleHeightChange = (newHeight: number | string) => {
+    if (iframe.current) {
+      iframe.current.style.height = `${newHeight}px`;
+    }
+  };
+
```

Comment:
I'd hope that duplicate code like this can be refactored when working on moving the interactive runtime api into typescript.